python/ray/actor.py

```diff
@@ -1817,3 +1914,42 @@ def exit_actor():
             f"{worker.mode}. Call this API inside an actor methods"
             "if you'd like to exit the actor gracefully."
         )
+
+
+def _handle_tensor_transport(dst_actor, args):
+    """Handle tensor transport between actors.
+
+    Args:
+        dst_actor: The target actor to receive tensors
+        args: List of arguments that may contain ObjectRefs
+    """
+    from ray.experimental.channel import ChannelContext
+
+    ctx = ChannelContext.get_current()
+
+    # TODO(kevin85421): Currently, GPU objects only support 1 communicator.
+    if len(ctx.communicators) != 1 or not args:
+        return
+
+    actor_id_to_rank = {}
+    for arg in args:
+        if not isinstance(arg, ray.ObjectRef):
+            continue
+
+        worker = ray._private.worker.global_worker
+        tensor_meta = worker.in_actor_object_refs.get(arg, None)
+        if tensor_meta is None:
+            continue
+
+        src_actor, tensor_meta = tensor_meta
+        if not actor_id_to_rank:
+            actor_id_to_rank = {
+                a._ray_actor_id: i for i, a in enumerate(ctx.communicators[0])
+            }
+        src_rank = actor_id_to_rank[src_actor._ray_actor_id]
+        dst_rank = actor_id_to_rank[dst_actor._ray_actor_id]
```

Comment:
Please also add checks that the src_rank and dst_ranks are in the dictionary.

Use ValueError instead of `assert`, same for the below check too.

Reply:
Updated https://github.com/ray-project/ray/pull/52938/commits/2d73d7c4478ba1a10ff6cd54d327a30af596c915